### ISSUE TITLE
Segmentation fault: incompatible of TersorRT version

### ISSUE THREAD

**Issue by masteryun** [Ghustwb/MobileNet-SSD-TensorRT#40]
**when I run the command: ./mobileNet
then I get the following errors:**
`attempting to open cache file ../../model/MobileNetSSD_deploy.caffemodel.1.tensorcache
loading network profile from cache...
createInference
The engine plan file is incompatible with this version of TensorRT, expecting 5.0.6.3got -2100382470.0.2.1852793695, please rebuild.
createInference_end
Bindings after deserializing:
Segmentation fault (core dumped)
`
But the requirement lists shows TensorRT4 is OK.

**It looks like I should install tensorRT 5 according to the error, but when I run dpkg -l | grep TensorRT
it shows:**
`...~/MobileNet-SSD-TensorRT$ dpkg -l | grep TensorRT
ii  graphsurgeon-tf                               5.0.6-1+cuda10.0                                arm64        GraphSurgeon for TensorRT package
ii  libnvinfer-dev                                5.0.6-1+cuda10.0                                arm64        TensorRT development libraries and headers
ii  libnvinfer-samples                            5.0.6-1+cuda10.0                                all          TensorRT samples and documentation
ii  libnvinfer5                                   5.0.6-1+cuda10.0                                arm64        TensorRT runtime libraries
ii  python-libnvinfer                             5.0.6-1+cuda10.0                                arm64        Python bindings for TensorRT
ii  python-libnvinfer-dev                         5.0.6-1+cuda10.0                                arm64        Python development package for TensorRT
ii  python3-libnvinfer                            5.0.6-1+cuda10.0                                arm64        Python 3 bindings for TensorRT
ii  python3-libnvinfer-dev                        5.0.6-1+cuda10.0                                arm64        Python 3 development package for TensorRT
ii  tensorrt                                      5.0.6.3-1+cuda10.0                              arm64        Meta package of TensorRT
ii  uff-converter-tf                              5.0.6-1+cuda10.0                                arm64        UFF converter for TensorRT package`

**My TensorRT is 5!
Someone can help me out with it? Thanks.**

**Comment by BRUCE11111** [comment]
I got the same errors as yours.Do you fix it?How? Thank you.

**Comment by masteryun** [comment]
nope. But I think it probably works after installing opencv, cudnn, and tensorrt via jetpack 3.3, which meets the environment requirements claimed by the author. 

发自我的 iPhone

**Comment by BRUCE11111** [comment]
Thanks for the reply.I found the cause of the problem.
These code must run with TensorRT4.The reason is that the code use serialized engine (tensorRT's format) which are not portable across platforms or TensorRT versions. 
If we use tensorRT4,it works fine.

**Comment by masteryun** [comment]
Cheers!

发自我的 iPhone

**Comment by ljdongysu** [comment]
When I build it with RT'6 ,it  turns out ‘cudaSoftmax(int, int, float*, float*)’ is not defined, delete cudaSoftmax function can build suddessfully,but running with error Assertion `C2 == inputDims[param.inputOrder[1]].d[0]' failed.How can deal with it ?Is it caused by RT version?

**Comment by masteryun** [comment]
TensorRT 4 is necessary for this project. hope this can help！

发自我的 iPhone

**Comment by huiofficial** [comment]
Looks like you are running with TensorRT 5.x, try to use 4.0. Hope this could help!
